java/tracing/src/main/scala/org/typelevel/otel4s/java/trace/SpanBackendImpl.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+package java
+package trace
+
+import cats.effect.Sync
+import cats.syntax.functor._
+import io.opentelemetry.api.trace.{Span => JSpan}
+import io.opentelemetry.api.trace.{StatusCode => JStatusCode}
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.meta.InstrumentMeta
+import org.typelevel.otel4s.trace.Span
+import org.typelevel.otel4s.trace.SpanContext
+import org.typelevel.otel4s.trace.Status
+
+import scala.concurrent.duration.FiniteDuration
+
+private[otel4s] class SpanBackendImpl[F[_]: Sync](
+    val jSpan: JSpan,
+    spanContext: SpanContext
+) extends Span.Backend[F] {
+  import SpanBackendImpl._
+
+  val meta: InstrumentMeta[F] = InstrumentMeta.enabled
+
+  def context: SpanContext =
+    spanContext
+
+  def addEvent(name: String, attributes: Attribute[_]*): F[Unit] =
+    Sync[F]
+      .delay(jSpan.addEvent(name, Conversions.toJAttributes(attributes)))
+      .void
+
+  def addEvent(
+      name: String,
+      timestamp: FiniteDuration,
+      attributes: Attribute[_]*
+  ): F[Unit] =
+    Sync[F]
+      .delay(
+        jSpan.addEvent(
+          name,
+          Conversions.toJAttributes(attributes),
+          timestamp.length,
+          timestamp.unit
+        )
+      )
+      .void
+
+  def setAttributes(attributes: Attribute[_]*): F[Unit] =
+    Sync[F]
+      .delay(jSpan.setAllAttributes(Conversions.toJAttributes(attributes)))
+      .void
+
+  def setStatus(status: Status): F[Unit] =
+    Sync[F].delay(jSpan.setStatus(toJStatus(status))).void
+
+  def setStatus(status: Status, description: String): F[Unit] =
+    Sync[F].delay(jSpan.setStatus(toJStatus(status), description)).void
+
+  def recordException(
+      exception: Throwable,
+      attributes: Attribute[_]*
+  ): F[Unit] =
+    Sync[F]
+      .delay(
+        jSpan.recordException(exception, Conversions.toJAttributes(attributes))
+      )
+      .void
+
+  private[otel4s] def end: F[Unit] =
+    Sync[F].delay(jSpan.end())
```

Comment:
Yes, once the underlying span is stopped, all subsequent calls (e.g. `setStatus`, `addAttribute`) will not have any effect.

Also, we need `jSpan` to manipulate the context: for example, when we need to attach a newly created span to a parent. OpenTelemetry tracing relies on the thread-local context. Since the current implementation is only a wrapper around java API, we need to mimic this behavior using [IOLocal](https://github.com/typelevel/otel4s/blob/9ff15db3ee11522b783d5e9dc8401396782011df/java/tracing/src/main/scala/org/typelevel/otel4s/java/trace/TraceScope.scala
) instead. 

For example, inside of the `makeScope`, the `TraceScope` will attach this span (`jSpan`) to a parent context.:
https://github.com/typelevel/otel4s/blob/9ff15db3ee11522b783d5e9dc8401396782011df/java/tracing/src/main/scala/org/typelevel/otel4s/java/trace/SpanBuilderImpl.scala#L166-L169

https://github.com/typelevel/otel4s/blob/9ff15db3ee11522b783d5e9dc8401396782011df/java/tracing/src/main/scala/org/typelevel/otel4s/java/trace/TraceScope.scala#L89-L107